provider/testing/resource.go

```diff
@@ -126,44 +128,71 @@ func getEnv(key, fallback string) string {
 	return fallback
 }
 
-func verifyNoEmptyColumns(t *testing.T, tc ResourceTestData, conn pgxscan.Querier) {
-	// Test that we don't have missing columns and have exactly one entry for each table
-	for _, table := range getTablesFromMainTable(tc.Table) {
-		query := fmt.Sprintf("select * FROM %s ", table)
-		rows, err := conn.Query(context.Background(), query)
-		if err != nil {
-			t.Fatal(err)
-		}
-		count := 0
-		for rows.Next() {
-			count += 1
-		}
-		if count < 1 {
-			t.Fatalf("expected to have at least 1 entry at table %s got %d", table, count)
-		}
-		if tc.SkipEmptyJsonB {
-			continue
-		}
-		query = fmt.Sprintf("select t.* FROM %s as t WHERE to_jsonb(t) = jsonb_strip_nulls(to_jsonb(t))", table)
-		rows, err = conn.Query(context.Background(), query)
-		if err != nil {
-			t.Fatal(err)
-		}
-		count = 0
-		for rows.Next() {
-			count += 1
-		}
-		if count < 1 {
-			t.Fatalf("row at table %s has an empty column", table)
-		}
-	}
+type VerificationSequence struct {
+	TableName      string
+	ForeignKeyName string
+	Relations      []VerificationSequence
 }
 
-func getTablesFromMainTable(table *schema.Table) []string {
-	var res []string
-	res = append(res, table.Name)
+func getVerificationSequence(table *schema.Table) VerificationSequence {
+	res := VerificationSequence{
+		TableName: table.Name,
+	}
+	for _, c := range table.Columns {
+		if strings.Contains(c.Name, "_cq_id") {
+			res.ForeignKeyName = c.Name
+		}
+	}
 	for _, t := range table.Relations {
-		res = append(res, getTablesFromMainTable(t)...)
+		res.Relations = append(res.Relations, getVerificationSequence(t))
 	}
 	return res
 }
+func verifyColumnsBySequence(t *testing.T, conn pgxscan.Querier, sequence VerificationSequence, parentID *string) {
+	sq := squirrel.StatementBuilder.
+		PlaceholderFormat(squirrel.Dollar).
+		Select(fmt.Sprintf("json_agg(%s)", sequence.TableName)).
+		From(sequence.TableName)
+	if parentID == nil {
+		// it is a root object
+		sq = sq.Where(squirrel.Eq{"account_id": "testAccount"})
```

Comment:
this seems AWS specific, how would this work on other providers?

Reply:
Yeah. That is a great suggestion.